x/wasmplus/handler.go

```diff
@@ -22,13 +20,11 @@ func NewHandler(k wasmtypes.ContractOpsKeeper) sdk.Handler {
 			res proto.Message
 			err error
 		)
-		res, err = wasmHandler(ctx, msg)
-		if err != nil && strings.Contains(err.Error(), "MsgStoreCodeAndInstantiateContract") {
-			// handle wasmplus service
-			msg2, ok := msg.(*types.MsgStoreCodeAndInstantiateContract)
-			if ok {
-				res, err = msgServer.StoreCodeAndInstantiateContract(sdk.WrapSDKContext(ctx), msg2)
-			}
+		switch msg := msg.(type) {
+		case *types.MsgStoreCodeAndInstantiateContract:
+			res, err = msgServer.StoreCodeAndInstantiateContract(sdk.WrapSDKContext(ctx), msg)
+		default:
+			return wasmHandler(ctx, msg)
```

Comment:
I've implemented this before because I wanted the origin wasm handler to fire first, followed by wasmplus' handler. Because most tx's are tx's from wasm handlers.

So what if we keep the existing code?

Reply:
I benchmarked the type-switching and handling x/wasm's message (StoreCode) by x/wasm's handle and x/wasmplus's handler.

Benchmark of type switching by `StoreCodeAndInstantiateContract` or not:

```
BenchmarkTypeSwitch-12          1000000000               0.3464 ns/op
BenchmarkNoTypeSwitch-12        1000000000               0.2764 ns/op
```

Benchmark of handling `StoreCode` message:
1st try
```
BenchmarkWasmHandler-12               20          60633371 ns/op
BenchmarkWasmPlusHandler-12           20          52084338 ns/op
```
2nd try
```
BenchmarkWasmHandler-12               21          57135565 ns/op
BenchmarkWasmPlusHandler-12           20          61114532 ns/op
```

Diffs from type switching is very very small. I think it can be ignored for code readability.